tests/LSP6KeyManager/internals/AllowedERC725YKeys.internal.ts

```diff
@@ -1,168 +1,728 @@
-import { expect } from "chai";
 import { ethers } from "hardhat";
-import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-
-// constants
-import {
-  SupportedStandards,
-  ERC725YKeys,
-  ALL_PERMISSIONS,
-  PERMISSIONS,
-} from "../../../constants";
+import { expect } from "chai";
 
-// setup
+import { BytesLike } from "ethers";
 import { LSP6InternalsTestContext } from "../../utils/context";
-import { setupKeyManagerHelper } from "../../utils/fixtures";
 
-// helpers
-import { abiCoder } from "../../utils/helpers";
+export type DataKey = {
+  length: BytesLike;
+  key: BytesLike;
+};
 
 export const testAllowedERC725YKeysInternals = (
   buildContext: () => Promise<LSP6InternalsTestContext>
 ) => {
-  let context: LSP6InternalsTestContext;
+  describe("Testing CheckAllowedERC725YKeys", () => {
+    let context: LSP6InternalsTestContext;
+    let dataKeys: {
+      firstDynamicKey: DataKey;
+      secondDynamicKey: DataKey;
+      thirdDynamicKey: DataKey;
+      fourthDynamicKey: DataKey;
+      firstFixedKey: DataKey;
+      secondFixedKey: DataKey;
+      thirdFixedKey: DataKey;
+      fourthFixedKey: DataKey;
+    };
+    let dataKeysToCheck: BytesLike[];
+    let compactBytesArray_2d: BytesLike;
+    let compactBytesArray_2f: BytesLike;
+    let compactBytesArray_2d_2f: BytesLike;
+    let compactBytesArray_mixed_d_f: BytesLike;
+    before(async () => {
+      context = await buildContext();
+      dataKeys = {
+        firstDynamicKey: {
+          length: "0x11",
+          key: "0xaf449139942203369080622073bf7f2dab",
+        },
+        secondDynamicKey: {
+          length: "0x01",
+          key: "0x1c",
+        },
+        thirdDynamicKey: {
+          length: "0x1c",
+          key: "0x4f042128e305e375c54b8782c3f9f1bde93f3586649d48db9c68beef",
+        },
+        fourthDynamicKey: {
+          length: "0x15",
+          key: "0xe6e8c1d23558e2a87caf19b7cc928eff323881d3e6",
+        },
+        firstFixedKey: {
+          length: "0x20",
+          key: "0xcbfb606f69bb97c04cbaea2a8b7cb13a2e229fafa3fb3be6db11d96ee3add114",
+        },
+        secondFixedKey: {
+          length: "0x20",
+          key: "0xa2c116feaaf87e499ac78081e8ce74b0a85627265144d605904971a89f81220a",
+        },
+        thirdFixedKey: {
+          length: "0x20",
+          key: "0x7b68176ed8c5774f16c8040df3f2c4aac9959612242613785716c4263670fe18",
+        },
+        fourthFixedKey: {
+          length: "0x20",
+          key: "0x2b58178172d258515ef1d9e7c467f6f6a09510e863ef5ad383dbfc50721183df",
+        },
+      };
+      dataKeysToCheck = [
```

Comment:
Not so important, but maybe to clarify:

```suggestion
      dataKeysToRevert = [
```